src/fields/fq2.rs

```diff
@@ -162,10 +162,16 @@ impl Mul for Fq2 {
         let prime: [u32; 8] = bytemuck::cast(Fq::modulus().0);
         let prime_sqr: [u32; 16] = bytemuck::cast(Fq::modulus_squared().0);
 
-        let mut result_mut = [[0u32; 8]; 2];
+        // We construct our result using u32s but will eventually need to interpret it as u128s, so
+        // we align to the u128 alignment (i.e. of 16 bytes)
+        #[repr(align(16))]
+        struct AlignedU32s {
+            val: [[u32; 8]; 2]
+        }
+        let mut result_mut = AlignedU32s{ val: [[0u32; 8]; 2] };
 
-        field::extfield_xxone_mul_256(&lhs, &rhs, &prime, &prime_sqr, &mut result_mut);
-        let result: &[[u128; 2]; 2] = bytemuck::cast_ref(&result_mut);
+        field::extfield_xxone_mul_256(&lhs, &rhs, &prime, &prime_sqr, &mut result_mut.val);
+        let result: &[[u128; 2]; 2] = bytemuck::cast_ref(&result_mut.val);
```

Comment:
Optionally, could be a newtype, if you prefer that syntax, sth like:

```
        // We construct our result using u32s but will eventually need to interpret it as u128s, so
        // we align to the u128 alignment (i.e. of 16 bytes)
        #[repr(align(16))]
        struct AlignedU32s([[u32; 8]; 2]);
        let mut result_mut = AlignedU32s([[0u32; 8]; 2]);

        field::extfield_xxone_mul_256(&lhs, &rhs, &prime, &prime_sqr, &mut result_mut.0);
        let result: &[[u128; 2]; 2] = bytemuck::cast_ref(&result_mut.0);
```

Reply:
oh yeah that does seem cleaner, thanks!